sessiontxn/failpoint.go

```diff
@@ -126,6 +132,27 @@ func TsoRequestCountInc(sctx sessionctx.Context) {
 	sctx.SetValue(TsoRequestCount, count)
 }
 
+// TsoWaitCountInc is used only for test
+// When it is called, there is a waiting tso operation
+func TsoWaitCountInc(sctx sessionctx.Context) {
+	count, ok := sctx.Value(TsoWaitCount).(uint64)
```

Comment:
For this use case, I think you'd better use an integer pointer and modify it atomically.